dotCMS/src/integration-test/java/com/dotmarketing/quartz/job/IntegrityDataGenerationJobTest.java

```diff
@@ -0,0 +1,168 @@
+package com.dotmarketing.quartz.job;
+
+import com.dotcms.IntegrationTestBase;
+import com.dotcms.integritycheckers.IntegrityUtil;
+import com.dotcms.publisher.endpoint.bean.PublishingEndPoint;
+import com.dotcms.publisher.endpoint.bean.impl.PushPublishingEndPoint;
+import com.dotcms.rest.IntegrityResource;
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.common.db.DotConnect;
+import com.dotmarketing.exception.DotDataException;
+import com.tngtech.java.junit.dataprovider.DataProviderRunner;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.quartz.JobDataMap;
+import org.quartz.JobDetail;
+import org.quartz.JobExecutionContext;
+import org.quartz.SchedulerException;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Properties;
+import java.util.UUID;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * @author victor
+ */
+@RunWith(DataProviderRunner.class)
+public class IntegrityDataGenerationJobTest extends IntegrationTestBase {
+
+    private IntegrityDataGenerationJob integrityDataGenerationJob;
+    private PublishingEndPoint endpoint;
+    private String requestId;
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
```

Comment:
Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6151243)